uffi-compat/uffi-compat.lisp

```diff
@@ -628,6 +632,12 @@ output to *trace-output*.  Returns the shell's exit code."
           `((with-foreign-strings ,(cdr bindings) ,@body))
           body)))
 
+#+ecl
+(defun foreign-string-length (foreign-string)
+  (declare (inline ffi:foreign-string-length))
+  (ffi:foreign-string-length foreign-string))
```

Comment:
OK, I'd remove it then. Can you rebase the PR with that change? (Maybe add a note about the run-shell-command fix, if you're feeling chatty. :-))